### PR TITLE
py-pandas: correct Python version constraint

### PR DESCRIPTION
**Pull request by adamjstewart**

### CODE REVIEW

**Review by rgommers — APPROVED** [review]
Thanks Adam! LGTM

File: var/spack/repos/builtin/packages/py-pandas/package.py [hunk 1]
```diff
@@ -97,7 +97,7 @@ class PyPandas(PythonPackage):
     with default_args(type=("build", "run")):
         # Based on PyPI wheel versions
         depends_on("python@3.9:3.13", when="@2.2.3:")
-        depends_on("python@3.9:3.12", when="@2.1.1:")
+        depends_on("python@3.9:3.12", when="@2.1.1:2.2.2")
```
Comment (haampie) [inline var/spack/repos/builtin/packages/py-pandas/package.py]: I would suggest splitting again:

```python
        depends_on("python@3.9:", when="@2.1:")
        depends_on("python@3.8:", when="@1.4:")

        # Based on PyPI wheel availability
        depends_on("python@:3.13")
        depends_on("python@:3.12", when="@:2.2.2")
        depends_on("python@:3.11", when="@:2.1.0")
        depends_on("python@:3.10", when="@:1.4")
        depends_on("python@:3.9", when="@:1.3.2")
        depends_on("python@:3.8", when="@:1.1.2")
```

wrt "Based on PyPI wheel availability" you just mean forward compat with python, right?

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-pandas/package.py]: By "Based on PyPI wheel availability" I mean "Version 2.2.3 has wheels for Python 3.9–3.13"

**Review by haampie — APPROVED** [review]

File: var/spack/repos/builtin/packages/py-pandas/package.py (hunk 1 above)
Comment (haampie) [reply var/spack/repos/builtin/packages/py-pandas/package.py]: okay yeah, i thought the lowerbound would come from pyproject.toml

**Pull request merged**